project/hackyourtemperature/src/components/CityDetails.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { IoCloseCircle } from 'react-icons/io5';
+
+const CityDetails = ({ city, clickHandler }) => {
+   return (
+      <div>
+         {' '}
+         {city.cod === 200 ? (
+            <li className="weather-detail" key={city.id} id={city.id}>
+               <span>
+                  {' '}
+                  <IoCloseCircle className="close-icon" id={city.id} onClick={clickHandler} />
```

Comment:
ids used on elements to style them, manipulate them, and sometimes to identify them, like in this case (not arguing about it's randomness here), but if you have the id, passing it directly will be much more efficient.
```suggestion
                  <IoCloseCircle className="close-icon" onClick={() => clickHandler(city.id)} />
```


And, clicking the `X` button results with opening the card, in which it shouldn't be, moving it outside the component (in your case) will result in not invoking the `Link` component, and therefore you will remove the intended element without opening it.